src/foam/nanos/controller/ApplicationController.js

```diff
@@ -172,30 +175,48 @@ foam.CLASS({
       self.clientPromise.then(function() {
         self
           .addClass(self.myClass())
-          .tag({class: 'foam.nanos.u2.navigation.TopNavigation'})
+          .start('div', null, self.topNavigation_$).end()
           .start('div').addClass('stack-wrapper')
             .tag({class: 'foam.u2.stack.StackView', data: self.stack, showActions: false})
-          .end();
+          .end()
+          .start('div', null, self.footerView_$).end();
+
+          // Sets up application view
+          self.topNavigation_.add(self.TopNavigation.create());
```

Comment:
Why is this done here rather than just done directly above?

Reply:
So you can replace them in setPortalView?

Reply:
topNavigation_ is undefined at that point.